bin/check-ssl-root-issuer.rb

```diff
@@ -0,0 +1,128 @@
+#! /usr/bin/env ruby
+#
+#   check-ssl-root-issuer
+#
+# DESCRIPTION:
+#   Check that a certificate is chained to a specific root certificate issuer
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#
+#   Check that a specific website is chained to a specific root certificate
+#       ./check-ssl-root-issuer.rb \
+#           -u https://example.com \
+#           -i "CN=DST Root CA X3,O=Digital Signature Trust Co."
+#
+# LICENSE:
+#   Copyright Jef Spaleta (jspaleta@gmail.com) 2020
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'openssl'
+require 'uri'
+require 'net/http'
+require 'net/https'
+
+#
+# Check root certificate has specified issuer name
+#
+class CheckSSLRootIssuer < Sensu::Plugin::Check::CLI
+  option :url,
+         description: 'Url to check: Ex "https://google.com"',
+         short: '-u',
+         long: '--url URL',
+         required: true
+
+  option :issuer,
+         description: 'An X509 certificate issuer name, RFC2253 format Ex: "CN=DST Root CA X3,O=Digital Signature Trust Co."',
+         short: '-i',
+         long: '--issuer ISSUER_NAME',
+         required: true
+
+  option :regexp,
+         description: 'Treat the issuer name as a regexp',
+         short: '-r',
+         long: '--regexp',
+         default: false,
+         boolean: true,
+         required: false
+
+  option :format,
+         description: 'optional issuer name format. Defaults to RFC2253. Allowed values: RFC2253, ONELINE, COMPAT',
+         short: '-f',
+         long: '--format FORMAT_VAL',
+         default: 'RFC2253',
+         required: false
+
+  def validate_opts
+    config[:issuer_format] = OpenSSL::X509::Name::RFC2253
+    config[:issuer_format] = OpenSSL::X509::Name::ONELINE if config[:format] == 'ONELINE'
+    config[:issuer_format] = OpenSSL::X509::Name::COMPAT if config[:format] == 'COMPAT'
+  end
+
+  def validate_issuer(cert)
+    issuer = cert.issuer.to_s(config[:issuer_format])
+    if config[:regexp]
+      ra = Regexp.new(config[:issuer].to_s)
```

Comment:
can we have a better var name? I'd prefer longer more descriptive vars over shaving a few chars.